src/net/sourceforge/kolmafia/persistence/ConcoctionDatabase.java

```diff
@@ -3332,4 +3266,114 @@ public int hashCode() {
       return hash;
     }
   }
+
+  public static class UsableConcoctions {
+    private Map<ConcoctionType, LockableListModel<Concoction>> usableMap = new TreeMap<>();
+
+    public UsableConcoctions() {
+      for (ConcoctionType type : ConcoctionType.values()) {
+        usableMap.put(type, new LockableListModel<>());
+      }
+    }
+
+    public void fill() {
+      this.clear();
+      Collection<Concoction> allConcoctions = ConcoctionPool.concoctions();
+      for (Map.Entry<ConcoctionType, LockableListModel<Concoction>> entry :
+          this.usableMap.entrySet()) {
+        entry
+            .getValue()
+            .addAll(
+                allConcoctions.stream()
+                    .filter(c -> c.type == entry.getKey())
+                    .collect(Collectors.toList()));
```

Comment:
might be able to use `Collectors.groupingBy` here to only run through the list once. Less important the less `fill` gets called.

Reply:
Done.